### PR TITLE
Complete tests for count intrinsics

### LINKED ISSUE #26: Simplify code for count intrinsics in next CBMC version
There are a couple of PRs that will add support for count builtins in CBMC. This will allow the count intrinsics code in RMC to be simplified and its regression tests to be enabled.

### PR DESCRIPTION
**Pull request by adpaco-aws**
### Description of changes: 

Adds more complete tests for count intrinsics. Removes the old tests that were based on concrete values. 

### Resolved issues:

Resolves #26 
Part of #727 

### Testing:

* How is this change tested? Existing regression + changes to tests.

* Is this a refactor change? No.

### Checklist
- [x] Each commit message has a non-empty body, explaining why the change was made
- [x] Methods or procedures are documented
- [x] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: tests/kani/Intrinsics/Count/cttz_nonzero_panic.rs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// kani-verify-fail
+
+// Check that `cttz_nonzero` fails if zero is passed as an argument
+
+#![feature(core_intrinsics)]
+use std::intrinsics::cttz_nonzero;
+
+// Call `cttz_nonzero` with an unconstrained symbolic argument
+macro_rules! test_cttz_nonzero {
```
Comment (celinval) [inline tests/kani/Intrinsics/Count/cttz_nonzero_panic.rs:11]: Why is this named nonzero?

File: tests/kani/Intrinsics/Count/ctlz_nonzero_panic.rs [hunk 2]
```diff
@@ -0,0 +1,25 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// kani-verify-fail
```
Comment (celinval) [inline tests/kani/Intrinsics/Count/ctlz_nonzero_panic.rs:3]: Won't this test pass if just one property fail?

**Review by celinval — APPROVED** [review]
Those are nice tests. I added minor comments.

File: tests/kani/Intrinsics/Count/ctlz.rs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        // Check that the stable version returns the same value
+        assert!(ctlz(var) as u32 == var.leading_zeros());
+    };
+}
+
+// Assert that the results of `ctlz` and `ctlz_nonzero` are the same if we
+// exclude zero
+macro_rules! test_ctlz_nonzero {
+    ($ty:ty) => {
+        let var_nonzero: $ty = 8;
```
Comment (zhassan-aws) [inline tests/kani/Intrinsics/Count/ctlz.rs:41]: Was this intended to be assigned to `kani::any()` instead of 8?

Comment (adpaco-aws) [reply tests/kani/Intrinsics/Count/ctlz.rs]: Yes, thanks for catching it! Fixed 😄

File: tests/kani/Intrinsics/Count/cttz_nonzero_panic.rs (hunk 1 above)
Comment (adpaco-aws) [reply tests/kani/Intrinsics/Count/cttz_nonzero_panic.rs:11]: Because it is testing the `cttz_nonzero` intrinsic.

**Pull request merged**